server/sonar-process/src/main/java/org/sonar/process/command/CommandFactoryImpl.java

```diff
@@ -56,9 +59,15 @@
   private final Props props;
   private final File tempDir;
 
-  public CommandFactoryImpl(Props props, File tempDir) {
+  public CommandFactoryImpl(Props props, File tempDir, System2 system2) {
     this.props = props;
     this.tempDir = tempDir;
+    String javaToolOptions = system2.getenv(ENV_VAR_JAVA_TOOL_OPTIONS);
+    if (javaToolOptions != null && !javaToolOptions.trim().isEmpty()) {
+      LoggerFactory.getLogger(CommandFactoryImpl.class)
+        .warn(ENV_VAR_JAVA_TOOL_OPTIONS + " is defined but will be ignored. " +
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2629)